### PR TITLE
Update dg

### PR DESCRIPTION
**Pull request by tanchinkiat99**
Updated DG with class, activity and sequence diagrams for Filter, Undo and Redo features.

### CODE REVIEW

File: docs/diagrams/UndoRedoFeatureActivityDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    Addressbook is returned;
+
+    if () then ([User decides to redo the previous command])
+        :User executes a "Redo" command;
+        :The previous command is executed again and the
+        state of Addressbook before the "Undo"
+        command is returned;
+    else ([else])
+   endif
+endif
```
Comment (DannyDakota) [inline docs/diagrams/UndoRedoFeatureActivityDiagram.puml:27]: I think you are missing another endif over here? 👍 
to join the if and and else branch of the original alternate branch

File: docs/diagrams/RedoSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+participant ":AddressBook" as AddressBook MODEL_COLOR
+end box
+[-> LogicManager : execute(redo)
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand(redo)
+activate AddressBookParser
+
+create RedoCommand
+AddressBookParser -> RedoCommand
```
Comment (DannyDakota) [inline docs/diagrams/RedoSequenceDiagram.puml:21]: Hello~~ to follow to our team's convention, you can include the :RedoCommand at the end of line 21 💯

**Review by DannyDakota — CHANGES_REQUESTED** [review]
Other than the final branching and following team's conventions, the diagrams were all really well implemented especially in such a short period of time. Good job chin kiat 💯

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -495,76 +507,114 @@ The sequence diagram below shows the interactions between objects during the exe
...
+
+#### Execution of undo feature
+
+The `undo` command is as follows:
+
+`undo`
+
+Calling this command undoes only the last executed add, edit or delete command, and can only be used after executing an add, edit, or delete command.
+
+This command cannot be used in succession to undo previously executed commands besides the last executed one.
```
Comment (jiale-c) [inline docs/DeveloperGuide.md:528]: Good that you alert the reader. If you would like, you can use icons to draw reader's attention.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -495,76 +507,114 @@ The sequence diagram below shows the interactions between objects during the exe
...
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+**Path Execution of Undo and Redo Features Activity Diagram is shown below:**
+![UndoRedoFeatureActivityDiagram](images/UndoRedoFeatureActivityDiagram.png)
+
+#### Implementation
+
+The proposed undo/redo mechanism is facilitated by `AddressBook`. On top of the current state of the `UniquePersonList` of `Person` objects, it stores two additional states, the history and the "original" state. Additionally, it implements the following operations:
+
+* `AddressBook#saveHistory()` — Saves the current persons list state in its history.
```
Comment (jiale-c) [inline docs/DeveloperGuide.md:551]: I like this way of presentation.

**Review by jiale-c — CHANGES_REQUESTED** [review]
LGTM. But maybe you can change the redo and undo sequence diagram to have a call name for object creation such as: RedoCommand().

**Pull request closed without merging**